data/lua/wml/objectives.lua

```diff
@@ -186,7 +194,7 @@ function wml_actions.objectives(cfg)
 		cfg = helper.parsed(cfg)
 	end
 
-	local sides = wesnoth.get_sides(cfg)
+	local cfg_sides = wesnoth.get_sides(cfg)
```

Comment:
I don't understand why this was changed...

Reply:
Either `sides` or `sides_configuration` (or `sides_cfg`  if you must) depending upon whether it's usage

Reply:
@CelticMinstrel it was changed because there was variable shadowing in the code. Honestly, I can keep it as-is, but it's considered bad code style by some guides.

Reply:
// Offtopic: There was another file when there were variable overwrites AND shadowing, which is just pure hell if you want correct code. But I wasn't in the mood of fixing the harder usage. Also, changing something very complex may yield re-factoring mistakes on it's own.

Reply:
Let me clarify further: I figured the intent was to avoid variable shadowing, but I didn't see the variable it was shadowing.

Reply:
Ah, np. Here is the full context: https://github.com/wesnoth/wesnoth/blob/master/data/lua/wml/objectives.lua#L196

Reply:
I'm looking at that and still not seeing any variable shadowing.

Reply:
`sides` is defined here: https://github.com/wesnoth/wesnoth/blob/0db553ef36bc7d4d1ec10a7521a2c061d9234805/data/lua/wml/objectives.lua#L189
and here: https://github.com/wesnoth/wesnoth/blob/0db553ef36bc7d4d1ec10a7521a2c061d9234805/data/lua/wml/objectives.lua#L193
with the same value on one execution branch, but with different values in another execution branch.

Reply:
Ah, I see it now. Okay.

Reply:
@CelticMinstrel modify_unit.lua, the variable being `cfg`. See for youself if it needs handling, I'm not sure (I'd be a bit scared to touch the code without need).

Reply:
Hmm, I can see a lot of shadowing in that file, but no overwrites (or maybe I don't quite understand what you mean by that).

I think what I'd do with that file is refactor it so that the local functions are not nested in the other function (which would mean promoting `unit_variable` to the global scope and `replace_mode` to a parameter).

Reply:
Indeed, two different types of shadowing, but no overwrites. I would personally avoid using globals and would pass info as arguments or return values, if needed.

I wouldn't personally mind nesting too much, if the widest scope has long inconvenient variable names (those you wouldn't use in nested functions). But that's a very particular programming style, I'm open to other thoughts on nesting.

Anyways, this all is a bit off-topic for this PR...

Reply:
EDIT: cleaned up comment ^

Reply:
done